### DISCUSSION TITLE
How to create Telegram integration?

### DISCUSSION THREAD

**Discussion by doctorixx** [netalertx/NetAlertX#774] [Q&A] [answered]
I didn't find the plugin I needed. Help please. How to set up this integration?
Thank you in advance

**Comment by doctorixx** [comment, marked as answer]
1) I run Apprise in docker container.
2) Enable Apprise publisher
3) Set Apprise host URL http://192.168.0.13:20000/notify/ (My docker host ip and port)
4) Apprise notification URL like `tgram://{bot_token}/{chat_id}` .
Start your bot. Chat id is yout telegram account id if u wont message for only you

Press SAVE 

Works!
![image](https://github.com/user-attachments/assets/1c3f990d-0a3d-42dc-93d1-45f9e9c9c5d9)

**Reply by HelloMihai** [reply]
this worked for me as well.... 

for anyone else looking to debug

test apprise telegram notifications
replace `<bottoken>` and `<chatid>` with yours

test telegram notification in browser
```
https://api.telegram.org/bot<bottoken>/sendMessage?chat_id=<chatid>&text=%40%40TEXT%40%40
```

test apprise notification in console (replace 192.168.1.2:9999  with your apprise ip and port)
```
curl -X POST -d '{"urls":"tgram://<bottoken>/<chatid>","body":"test body from curl","title":"test title from curl"}' -H "Content-Type: application/json" "http://192.168.1.2:9999/notify/"
```

test from the docker apprise container console
```
apprise -vv -t "Test Message from apprise console" -b "Test Message from apprise console" \
   tgram://<bottoken>/<chatid>/
```

my netalertx config looks like this:
![image](https://github.com/user-attachments/assets/ea6adbdb-e9bb-4df7-84dc-b8019ba5a1a4)

**Reply by JohnyBeGood** [reply]
Screenshot helps alot. Thanks!

**Comment by jcxhard** [comment]
Hello friends, I was unable to integrate Apprise into my Telegram, I followed exactly the manual instructions. Could you help me?

**Reply by jokob-sk** [reply]
Hi, please share your settings and if you tried what is mentioned above:

> test apprise telegram notifications
> replace <bottoken> and <chatid> with yours
> 
> test telegram notification in browser
> 
> https://api.telegram.org/bot<bottoken>/sendMessage?chat_id=<chatid>&text=%40%40TEXT%40%40
> 
> test apprise notification in console (replace 192.168.1.2:9999 with your apprise ip and port)
> 
> curl -X POST -d '{"urls":"tgram://<bottoken>/<chatid>","body":"test body from curl","title":"test title from curl"}' -H "Content-Type: application/json" "http://192.168.1.2:9999/notify/"
> 
> test from the docker apprise container console
> 
> apprise -vv -t "Test Message from apprise console" -b "Test Message from apprise console" \
>    tgram://<bottoken>/<chatid>/

**Reply by jcxhard** [reply]
Ola amigo, tudo bem? Sim realizei todas as configurações

Segui os passos;

1. Criar o bot, para criar o bot_token e chat_id
https://github.com/caronc/apprise/wiki/Notify_telegram.

2. Instalei o Apprise API dentro do meu docker rodando na distribuição debian 12
https://hub.docker.com/r/caronc/apprise

3. Criei o ID no Apprise
http://192.168.32.148:8000/cfg/240bdccda636f84b44ead45eb6fd74012680e768bdd57f6b992d625af15940aa?theme=light
![image](https://github.com/user-attachments/assets/4c3914fa-7895-4be6-8f8d-b10168b23ab7)

4.  Segui essas configuraçoes conforme manual
![image](https://github.com/user-attachments/assets/c3485297-daa1-401a-8959-01688c99f61c)

--------------------

Os testes que voce me pediu para realizar

1. Teste de notificação do telegram

![image](https://github.com/user-attachments/assets/ab7251da-9fea-4514-ada4-ef6c17e799f8)

2.  Teste de notificação no console do meu docker, não me retornou nenhum erro.

root@docker-srv01:~# curl -X POST -d '{"urls":"tgram:///","body":"test body from curl","title":"test title from curl"}' -H "Content-Type: application/json" "https://192.168.32.148:8000/notify/"

3.  Teste de console do meu conteiner docker
4. 
![image](https://github.com/user-attachments/assets/c2e7a58b-9588-4a17-bcc3-a14929739975)

Simplestente não consigo receber notificaçoes no meu Bot do Telegram. Faltou algo a fazer?

**Reply by jcxhard** [reply]
Corrigindo, 

a seguencia de comando no meu teste seria

curl -X POST -d '{"urls":"tgram:///","body":"test body from curl","title":"test title from curl"}' -H "Content-Type: application/json" "http://192.168.32.148:8000/notify/"

**Reply by doctorixx** [reply]
> Hello friends, I was unable to integrate Apprise into my Telegram, I followed exactly the manual instructions. Could you help me?

Hi. You can use telegram publisher in netalertx without apprise

**Reply by jokob-sk** [reply]
@doctorixx - good callout!
@jcxhard  FYI, as per description, `APPRISE_PAYLOAD` needs to be set to `text` for telegram - your settings say incorrectly `html`
![apprise_telegram](https://github.com/user-attachments/assets/6c794883-86ca-47c6-830a-324375626114)

**Reply by jcxhard** [reply]
Hello friend, it worked, thank you very much!!!

**Reply by jcxhard** [reply]
@doctorixx - Hello friend, I tested your method and it worked too, thank you!!!

**Comment by jcxhard** [comment]
Hello friends, how do I update net-alertx through my docker, I'm using Portainer to manage my containers

![image](https://github.com/user-attachments/assets/392c182e-e4aa-4cbf-aeb8-41350281664f)

**Reply by jokob-sk** [reply]
Hi, you can find relevant docs here: https://jokob-sk.github.io/NetAlertX/UPDATES/?h=upgrad

Locking this discussion as we are getting off topic - feel free to open a new discussion on updates/upgrades
